From: Mikael Magnusson <mikma264@gmail.com>
Subject: [PATCH] t/fix/init_start

Start applications using -s switch to erl instead of using custom
boot file.

Signed-off-by: Mikael Magnusson <mikma@users.sourceforge.net>

---
 scripts/init.sh.in        |    2 +-
 src/appserver.erl         |   10 ++++++++++
 src/event/eventserver.erl |   10 ++++++++++
 src/incomingproxy.erl     |   10 ++++++++++
 src/outgoingproxy.erl     |   11 +++++++++++
 src/pstnproxy.erl         |   10 ++++++++++
 6 files changed, 52 insertions(+), 1 deletions(-)

diff --git a/scripts/init.sh.in b/scripts/init.sh.in
index 305f74a..8d0c3ae 100644
--- a/scripts/init.sh.in
+++ b/scripts/init.sh.in
@@ -320,7 +320,7 @@ else
 		    fi
 		fi
 	    fi
-	    ${erl} -boot "$bootfile" -pz ${beamdir} \
+	    ${erl} $boot_start_ssl -s ${PROGRAMNAME} -pz ${beamdir} \
 		$EXTRA_ERL_ARGUMENTS \
 		-name $name \
 		-yxa_config $configpath \
diff --git a/src/appserver.erl b/src/appserver.erl
index 43f30e8..66758d1 100644
--- a/src/appserver.erl
+++ b/src/appserver.erl
@@ -24,6 +24,11 @@
 
 -export([test/0]).
 
+-export([
+	 start/0,
+	 stop/0
+	]).
+
 %%--------------------------------------------------------------------
 %%% Application specific exports, exported for CPL subsystem
 %%--------------------------------------------------------------------
@@ -49,6 +54,11 @@
 -define(APPSERVER_GLUE_TIMEOUT, 1200 * 1000).
 -define(SIPPIPE_TIMEOUT, 900).
 
+start() ->
+    application:start(?MODULE).
+
+stop() ->
+    application:stop(?MODULE).
 
 %%====================================================================
 %% Behaviour functions
diff --git a/src/event/eventserver.erl b/src/event/eventserver.erl
index fc451ca..ba6cff6 100644
--- a/src/event/eventserver.erl
+++ b/src/event/eventserver.erl
@@ -23,6 +23,11 @@
 
 -export([test/0]).
 
+-export([
+	 start/0,
+	 stop/0
+	]).
+
 %%--------------------------------------------------------------------
 %%% Application specific exports
 %%--------------------------------------------------------------------
@@ -43,6 +48,11 @@
 -define(EVENTSERVER_T, eventserver_t).
 -define(SHUTDOWN_TIMEOUT, 10).
 
+start() ->
+    application:start(?MODULE).
+
+stop() ->
+    application:stop(?MODULE).
 
 %%====================================================================
 %% Behaviour functions
diff --git a/src/incomingproxy.erl b/src/incomingproxy.erl
index 511bd66..5fc8aa9 100644
--- a/src/incomingproxy.erl
+++ b/src/incomingproxy.erl
@@ -25,6 +25,11 @@
 
 -export([test/0]).
 
+-export([
+	 start/0,
+	 stop/0
+	]).
+
 %%--------------------------------------------------------------------
 %% Include files
 %%--------------------------------------------------------------------
@@ -32,6 +37,11 @@
 -include("sipsocket.hrl").
 -include("yxa_config.hrl").
 
+start() ->
+    application:start(?MODULE).
+
+stop() ->
+    application:stop(?MODULE).
 
 %%====================================================================
 %% Behaviour functions
diff --git a/src/outgoingproxy.erl b/src/outgoingproxy.erl
index a5ce107..d2bf5ca 100644
--- a/src/outgoingproxy.erl
+++ b/src/outgoingproxy.erl
@@ -27,6 +27,11 @@
 
 -export([test/0]).
 
+-export([
+	 start/0,
+	 stop/0
+	]).
+
 %%--------------------------------------------------------------------
 %% Include files
 %%--------------------------------------------------------------------
@@ -34,6 +39,12 @@
 -include("sipsocket.hrl").
 -include("yxa_config.hrl").
 
+start() ->
+    application:start(?MODULE).
+
+stop() ->
+    application:stop(?MODULE).
+
 %%====================================================================
 %% Behaviour functions
 %% Standard YXA SIP-application callback functions
diff --git a/src/pstnproxy.erl b/src/pstnproxy.erl
index c5ebd75..5218403 100644
--- a/src/pstnproxy.erl
+++ b/src/pstnproxy.erl
@@ -28,6 +28,11 @@
 
 -export([test/0]).
 
+-export([
+	 start/0,
+	 stop/0
+	]).
+
 %%--------------------------------------------------------------------
 %% Include files
 %%--------------------------------------------------------------------
@@ -36,6 +41,11 @@
 -include("pstnproxy.hrl").
 -include("yxa_config.hrl").
 
+start() ->
+    application:start(?MODULE).
+
+stop() ->
+    application:stop(?MODULE).
 
 %%====================================================================
 %% Behaviour functions
-- 
tg: (5aeee79..) t/fix/init_start (depends on: upstream)
